src/zerocoin_v3.cpp

```diff
@@ -471,12 +462,7 @@ bool ConnectBlockZCV3(
 				zerocoinStateV3.AddSpend(serial.first);
 			}
 		}
-        // Shows if V3 sigma mints are now allowed.
-        bool V3MintsAllowed = (pindexNew->nHeight >= Params().GetConsensus().nSigmaStartBlock);
```

Comment:
Can you describe me how this was causing the sync problems? I'm just curious to know.

Reply:
This is exactly the problem I warned about. If block gets disconnected and gets connected again and mints are not cleared they will be doubled

Reply:
Once the block header is downloaded CBlockIndex is retained and reused if referenced later. Even in case when the block is disconnected

Reply:
Nope this one is not the problems. But rejecting sigma mint before activation block is not the right way as we discussed before so I remove it. We don't need to check here because `pblock->zerocoinTxInfoV3->mints` will always empty if sigma is not activated yet.

Reply:
I'm pretty sure I was writing comments to different line of code and now it's here. Weird